### PR TITLE
Wrap ether refactor

### PR DESCRIPTION
**Pull request by anxolin**
This is the first of 2 PRs for adding the Wrap/Unwrap functionality.
* This PR: All UX, and components
* Next PR: New `WethApi` and hook that into the components of this PR

----

I would love a review of the PR, and ideally some help figuring out what I wrote in the chat. 
I don't understand why if use `onSubmit` in the form, although I prevent the default, the form is being submitted. :cry:

Currently, I removed (commented out) the form logic and now I use `watch` of `useForm`, but I also have an issue with that as you might see when you test this ---> you need to clicks to send the form.

I think I can fix this by not using `useForm` and using just a controlled form input, but probably there's sth I'm missing and one of you might help me understand :) 

----

This PR includes some new components in `WrapEtherBtn.tsx`
- `WrapEtherBtn`
- `UnwrapEtherBtn`

They are a button that opens a the wrap/unwrap modal. So the modal is there, although not exposed as a component.

Then, in the trade page, we use the wrap button, only when the user selects WETH and the token is the sell token:
```js
<WrapEtherBtn label="+ Wrap Ether" />
```

image

If you click the button, it'll open the modal:
image

Also, for the balance, I've added the WRAP/UNWRAP for the row of the WETH token:
```js
<WrapEtherBtn label="Wrap" className="wrapUnwrapEther" />
<UnwrapEtherBtn label="Unwrap" className="wrapUnwrapEther" />
```
image

As a bonus, it does a small refactor of `ModalBodyWrapper`, `InputBox ` and some defaults that the modals were using, since it was needed for this modal and it was coupled to the deposit widget.

Also, I had to change the z-index of the tooltip, because it was being shown behind the highlighted row in balances, and behind the modals too.

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr840--dexreact.review.gnosisdev.com

File: src/components/WrapEtherBtn.tsx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+      <Modali.Button label="Cancel" key="no" isStyleCancel onClick={(): void => toggleModal()} />,
+      <Modali.Button
+        label="Continue"
+        key="yes"
+        isStyleDefault
+        onClick={(): void => {
+          // if (formRef.current) {
+          //   formRef.current.submit()
+          // }
+          // handleSubmit()
```
Comment (Velenir) [inline src/components/WrapEtherBtn.tsx:250]: You should be able to do `handleSubmit(onSubmit())()`
https://react-hook-form.com/api/#handleSubmit

Comment (Velenir) [reply src/components/WrapEtherBtn.tsx:250]: `handleSubmit(onSubmit())()()()()()` :laughing: :rofl: :upside_down_face: :laughing:

File: src/components/TradeWidget/TokenRow.tsx [hunk 2]
```diff
@@ -275,6 +231,7 @@ const TokenRow: React.FC<Props> = ({
               + Deposit
             </TooltipWrapper>
           )}
+          {!readOnly && selectedToken.addressMainnet === WETH_ADDRESS_MAINNET && <WrapEtherBtn label="+ Wrap Ether" />}
```
[the remark this answers is not in the document]
Comment (W3stside) [reply src/components/TradeWidget/TokenRow.tsx:234]: yeah, any reason for no rinkeby wrapping? i like testing there

File: src/components/WrapEtherBtn.tsx [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        display: flex;
+        flex-flow: row wrap;
+
+        i {
+          position: absolute;
+          right: 1rem;
+          top: 0;
+          bottom: 0;
+          margin: 0;
+          color: #476481;
```
Comment (W3stside) [inline src/components/WrapEtherBtn.tsx:31]: can you use `var(--blah-blah)` here please? for dark mode compat etc.

File: src/components/WrapEtherBtn.tsx [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+          font-style: normal;
+          display: flex;
+          justify-content: center;
+          align-items: center;
+        }
+      }
+    }
+
+    .error {
+      color: red;
```
Comment (W3stside) [inline src/components/WrapEtherBtn.tsx:46]: also here - this is the eye-bleeding-on-dark-mode colour red

File: src/components/WrapEtherBtn.tsx [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+    }
+
+    b {
+      font-size: 1.3rem;
+      color: #2f3e4e;
```
Comment (W3stside) [inline src/components/WrapEtherBtn.tsx:51]: ditto

File: src/components/WrapEtherBtn.tsx [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+  className?: string
+}
+
+export type WrapEtherBtnProps = Omit<WrapUnwrapEtherBtnProps, 'wrap'>
+
+interface WrapUnwrapInfo {
+  title: string
+  symbolSource: string
+  balance: BN
+  tooltipText: JSX.Element | string
```
Comment (W3stside) [inline src/components/WrapEtherBtn.tsx:86]: I think it's recommended to use `React.ReactNode` over `JSX.Element` (also below)

File: src/components/WrapEtherBtn.tsx [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+
+function getModalParams(
+  wrap: boolean,
+  wethHelpVisible: boolean,
+  showWethHelp: React.Dispatch<React.SetStateAction<boolean>>,
+): WrapUnwrapInfo {
+  const WethHelp = (
+    <div className="more-info">
+      <p>
+        Gnosis Protocol allows to exchange any ERC20 token, but since ETH is not an ERC20, it needs to be wrapped first.
```
Comment (W3stside) [inline src/components/WrapEtherBtn.tsx:99]:
```suggestion
        Gnosis Protocol supports the trading of any ERC20 token. As ETH is not an ERC20 token, it must first be wrapped.
```

File: src/components/WrapEtherBtn.tsx [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+      </p>
+      <p>
+        Wrapping is sending your ether into the WETH contract. That would mint the same amount of WETH as ETH you send
```
Comment (W3stside) [inline src/components/WrapEtherBtn.tsx:102]:
```suggestion
        By wrapping ETH you will be minting your submitted amount as WETH.
```

File: src/components/WrapEtherBtn.tsx [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+        to it.
+      </p>
+      <p>You can exchange back ETH for WETH any time, this is called Unwrapping WETH.</p>
```
Comment (W3stside) [inline src/components/WrapEtherBtn.tsx:105]:
```suggestion
      <p>ETH can be <b>wrapped</b> as WETH anytime. Equally, WETH can be <b>unwrapped</b> back into ETH</p>
```

File: src/components/WrapEtherBtn.tsx [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+          <a onClick={(): void => showWethHelp(!wethHelpVisible)}>
+            {wethHelpVisible ? '[-] Show less...' : '[+] Show more...'}
+          </a>
+        </p>
+        {wethHelpVisible && WethHelp}
+      </>
+    )
+    const tooltipText = (
+      <div>
+        Wrap converts ETH into WETH,
```
Comment (W3stside) [inline src/components/WrapEtherBtn.tsx:130]:
```suggestion
        Wrapping converts ETH into WETH,
```

File: src/components/WrapEtherBtn.tsx [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
+      symbolSource: 'ETH',
+      balance,
+      description,
+      tooltipText,
+    }
+  } else {
+    const description = (
+      <>
+        <p>
+          Unwrap converts back WETH into ETH.{' '}
```
Comment (W3stside) [inline src/components/WrapEtherBtn.tsx:151]:
```suggestion
          Unwrapping converts WETH back into ETH.{' '}
```

File: src/components/WrapEtherBtn.tsx [hunk 12]
```diff
@@ -0,0 +1,269 @@
...
+          {!wethHelpVisible && <a onClick={(): void => showWethHelp(true)}>Learn more...</a>}
+        </p>
+        {wethHelpVisible && WethHelp}
+      </>
+    )
+    // TODO: Get WETH balance
+    const balance = new BN('1234567800000000000')
+
+    return {
+      title: 'Unwrap Ether',
```
Comment (W3stside) [inline src/components/WrapEtherBtn.tsx:161]: I think you mean `Wrapped Ether` here (as it says Unwrap)

File: src/components/WrapEtherBtn.tsx (hunk 1 above)
Comment (Velenir) [reply src/components/WrapEtherBtn.tsx]: @W3stside  did exactly that :+1:

File: src/components/WrapEtherBtn.tsx [hunk 13]
```diff
@@ -0,0 +1,269 @@
...
+      amountLabel: 'Amount to Wrap',
```
Comment (W3stside) [inline src/components/WrapEtherBtn.tsx:162]: `Amount to Unwrap`?

File: src/components/WrapEtherBtn.tsx [hunk 14]
```diff
@@ -0,0 +1,269 @@
...
+      symbolSource: 'WETH',
+      balance,
+      description,
+      tooltipText: 'Unwrap converts back WETH into ETH',
```
Comment (W3stside) [inline src/components/WrapEtherBtn.tsx:166]:
```suggestion
      tooltipText: 'Unwrapping converts WETH back into ETH',
```

File: src/components/TradeWidget/TokenRow.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (anxolin) [reply src/components/TradeWidget/TokenRow.tsx]: Yes, wrapping in Rinkeby also, if you see closely, I'm evaluating with `addressMainnet` so I don't need to do any logic of checking in which network I am currently.

That is the address we use for the images, and points to mainnet always

File: src/components/WrapEtherBtn.tsx (hunk 3 above)
Comment (anxolin) [reply src/components/WrapEtherBtn.tsx:31]: Okay, I think I didn't modify any of this CSS. Anyways, is good advice

File: src/components/DepositWidget/Row.tsx [hunk 15]
```diff
@@ -115,8 +117,23 @@ export const Row: React.FC<RowProps> = (props: RowProps) => {
...
         <td data-label="Wallet" title={formatAmountFull({ amount: walletBalance, precision: decimals }) || ''}>
-          {(claiming || depositing) && spinner}
-          {formatAmount(walletBalance, decimals)}
+          {isWeth ? (
+            <ul>
+              <li>
+                0.1 ETH <WrapEtherBtn label="Wrap" className="wrapUnwrapEther" />
+              </li>
+              <li>
+                {(claiming || depositing) && spinner}
```
[the remark this answers is not in the document]
Comment (anxolin) [reply src/components/DepositWidget/Row.tsx:126]: Good idea, I'll open an issue for that

**Pull request merged**